### ISSUE TITLE
Add reload action

### ISSUE THREAD

**Issue by bjarnef** [rhythmagency/formulate#52]
In context menu (right-click menu) for tree nodes I think it should have added "Reload" action.

Furthermore change folder icons in actions:
* Add/create (e.g. create form) `icon-add`
* Delete (form) `icon-delete`
* Move (form) `icon-enter`
* Copy (form) `icon-documents` - if this is implemented at some point.

**Comment by Nicholas-Westby** [comment]
Definitely been meaning to change the action icons. Though, I usually poke around on icomoon rather than just stick to the built-in icons in Umbraco.

Why would a reload action be necessary? You can reload the nodes by refreshing the browser. I think it will even reselect whichever node you had selected before the refresh.

**Comment by bjarnef** [comment]
Yes, it is also great to add own custom icons, but for actions like these create, delete, move I think it would bedre best for consistency to use similar icons as in core  (as Umbraco Forms use too).

Yes, but that takes longer time and might collapse the tree structure.. again consistency.

**Comment by axiom12** [comment]
Hi Nicholas,

On my local instance I updated the icons in the flyout menu icons to match the icon of the item in the tree (each folder type uses a different icon) and to also use the standard Umbraco move and delete icons. But then I started thinking about it more. I made the first design choice because using the generic add (plus sign) icon would be ambiguous in the case where you could add a folder as well as a specific item (form, layout, etc.).

My other thought was to use the "folder-plus" icon in the icomoon set for folder adds and use the standard "plus" add icon for all the different add item actions. The trade off for better consistency in the action menu would be the created folder item icons (formulate-data-group, etc) being different than the icon pictured in the action menu (folder-plus). Then of course the folder item icons could be all made consistent with a standard folder icon to match the menu icon.

I don't know how strict your UX and design standards are. I'm open to whatever you think works best. Do you have any thoughts on this?

Thanks!

**Comment by Nicholas-Westby** [comment]
@axiom12 I don't entirely follow. Some screenshots might help.

Also, if you have some IcoMoon icons in mind, let me know which you're thinking for which actions and I can give the yay/nay.

**Comment by axiom12** [comment]
The first image is of the action menu for the Forms node of the Formulate section as I have modified it locally:

![image](https://cloud.githubusercontent.com/assets/1196759/19777221/300e9008-9c45-11e6-9664-7e04fae34533.png)

It seems unclear visually to use the standard create icon (plus - see below) for both create actions.

![image](https://cloud.githubusercontent.com/assets/1196759/19777296/88f07772-9c45-11e6-9813-c8726c068f1b.png)

So instead of the standard plus icon, I substituted the icon that is used on the actual item when it is created.

Another solution would be to make the tree respond like the content section does with document types. Clicking on the ellipsis shows a list of the items that can be created like below:

![image](https://cloud.githubusercontent.com/assets/1196759/19777372/eb6a5346-9c45-11e6-8e77-6fcb865d3931.png)

and right-clicking on the item opens the a menu with the standard create icon, which when clicked opens the list of available doc types, which for formulate and the form section would be Form and Folder

![image](https://cloud.githubusercontent.com/assets/1196759/19777425/1c8ada04-9c46-11e6-9881-d2ac66f97331.png)

I don't know if you want the UI refactored like this or if I should use my simple first solution, albeit inconsistent with the rest of the backoffice UI.

**Comment by axiom12** [comment]
Also because there is a different icon for each folder type I had to add some logic to show the correct icon in the menu based on what section you are in. It might be simpler to have a consistent container icon if they are simple containers with no specific functionality per section.

**Comment by axiom12** [comment]
I added the code to do this and submitted a pull request. The updates include icon updates and the refresh action on all menus.

**Comment by Nicholas-Westby** [comment]
Thanks for the extra details! Just a heads up that I won't be able to properly dig into this for a few weeks. However, here are my initial thoughts.

I'd like to keep the icons consistent for actions between Formulate trees (e.g., the "Add Form" icon would be the same as the "Add Validation" icon). Also, the icon you have linked to for the "Add Folder" icon seems like a good choice. Using the built-in icons for create/move/delete also seem reasonable.

**Comment by axiom12** [comment]
Ok I need to clean up a few things in the code then. I appreciate the feedback. It keeps me moving in the right direction. Do I need to remove the pull request or is rejecting it something you do?

**Comment by Nicholas-Westby** [comment]
I went ahead and closed it. By the way, I created a separate ticket for the icon updates: https://github.com/rhythmagency/formulate/issues/65

That way, a pull request can be submitted per issue (one for the reload action, and one for the icon changes).

**Comment by axiom12** [comment]
ok cool I'll make a new branch. thanks!
